share/spice/hackage/hackage.js

```diff
@@ -42,17 +42,15 @@
         var results = api_result.results;
 
         // Get the original query
-        var script = $('[src*="/js/spice/hackage/"]')[0];
-        var source = $(script).attr("src");
-        var query = source.match(/hackage\/([^\/]*)/)[1];
+        var query = DDG.get_query().replace(/\s*(hoogle|haskell|hackage)\s*/gi, '');
```

Comment:
It would be simplest to do `searchTerm: decodeURIComponent(query)`

This way no other changes are necessary 👍